### PR TITLE
Docker general and practical

### PR DESCRIPTION
**Pull request by eglerean**
In _episodes/05-environments.md I have added some practical usage of other's docker images (e.g. to run Rstudio without installing it or to run older version of anaconda without touching local install)

In _episodes/07-docker.md I have edited the Dockerfile example to reflect the changes that were needed to make it build. I have also added at the end some examples to push to dockerhub but also to store the image locally into a tar file (and how to reload it) when the content is sensitive.

### CODE REVIEW

File: _episodes/01-motivation.md [hunk 1]
```diff
@@ -67,6 +67,8 @@ domains of science:
 
 ## Reproducible, replicable, robust, generalisable
 
+While reproducibility is the minimum requirement and can be solved with "good enough" computational practices, replicability/robustness/generalisability of scientific findings are an even greater concern involving research misconduct, questionable research practices (p-hacking, HARKing, cherry-picking), sloppy methods, and other conscious and unconscious biases.
```
Comment (bast) [inline _episodes/01-motivation.md]: I just learned what HARKing means :-)

File: _episodes/07-docker.md [hunk 2]
```diff
@@ -107,17 +107,18 @@ LABEL maintainer="kthw@kth.se"
 # update the apt package manager
 RUN apt-get update
 RUN apt-get install -y software-properties-common
```
[the remark this answers is not in the document]
Comment (joasilva) [reply _episodes/07-docker.md]: > > These changes make the building of this image more efficient. But, two questions:
> > 
> > * Why are we using the `ubuntu:16.04` images, which is over 200MB, and not Alpine, which is a mere 5.55MB?
> > * Why are we using a private repo (`ppa:deadsnakes`) for installing python-3.6, and not an official repo?
> 
> Shouldn't we suggest to use an official python docker image for python? https://hub.docker.com/_/python
> 
> Or a conda image?

Good point. Something like, for example: https://github.com/docker-library/python/blob/a7db8db3c0eb92c86fbe74a91cb0eab4f7699a21/3.9-rc/alpine3.10/Dockerfile

**Comment by rkdarst** [comment]
Should we work on getting this merged?  Looks like some minor revisions to do, but otherwise there are no objections.

Maybe after Thursday, since it's being given tomorrow (=today).

**Comment by blankdots** [comment]
Providing my 2 cents on this, I think @annefou:
>  Shouldn't we suggest to use an official python docker image for python? https://hub.docker.com/_/python
> Or a conda image?

In practice the official images are more efficient. 

I would not install my own python version in an image unless I really need to, and even then I would look into multi-stage builds in order to optimise the image size. 

Portability is a key factor with Container images and that has some correlation with size.

**Comment by rkdarst** [comment]
So, this is hanging around for a while, and it seems that the blockers above were already in the Dockerfile, and not related to this PR.  I propose that we accept this PR, and then an issue for:

- Inherit from a different Dockerfile
- combine RUNs into single layers.

... though one could argue that the less smart way is better pedagogically, I don't think it applies too much here.

**Comment by wikfeldt** [comment]
i agree. This PR has two approvals, so i'll merge and will raise two new issues as Richard said

**Pull request merged**